src/core/manager.c

```diff
@@ -3317,6 +3319,62 @@ const char *manager_get_runtime_prefix(Manager *m) {
                getenv("XDG_RUNTIME_DIR");
 }
 
+int manager_set_exec_params(Manager *m, ExecParameters *p) {
+        int r;
+
+        assert(m);
+        assert(p);
+
+        p->environment = m->environment;
+        p->confirm_spawn = manager_get_confirm_spawn(m);
+        p->cgroup_supported = m->cgroup_supported;
+
+        if (MANAGER_IS_SYSTEM(m)) {
+                r = sd_path_home(SD_PATH_SYSTEM_RUNTIME, NULL, &p->prefix[EXEC_DIRECTORY_RUNTIME]);
```

Comment:
instead of invoking these calls each time we fork something of I think we should simple cache the prefixes in the `Manager` object (inside an array sized by _EXEC_DIRECTORY_MAX), and just call these functions once for each type at the time we initialize PID 1. That way we always have the paths ready, and we can just point the ExecParameter field to it, and thus don't need an exec_params_destroy function!